x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -178,6 +180,13 @@ public Realm realm(String name) {
                 }
                 internalTypes.add(type);
             }
+            if (KerberosRealmSettings.TYPE.equals(type)) {
+                if (isKerberosRealmConfigured) {
+                    throw new IllegalArgumentException(
+                            "multiple [" + type + "] realms are configured. [" + type + "] can only have one such realm configured");
+                }
+                isKerberosRealmConfigured = true;
+            }
```

Comment:
When lookup-realms is merged, we'll have the `initialize` method available which would allow us to move this  into the realm itself which would be a bit neater.

Reply:
The not-yet-merged `initialize` method gets passed a list of all the configured realms (which includes access to their type). A Kerberos realm can use that metohd to check that it is the only realm with `type == "kerberos"` and throw an exception is there are multiple such realms.

That would allow the _kerberos has special requirements_ logic to be contained within the Kerberos realm.